src/plots/cartesian/set_convert.js

```diff
@@ -103,7 +103,15 @@ module.exports = function setConvert(ax, fullLayout) {
                 // same logic as in Lib.ms2DateTime
                 var msecTenths = Math.floor(Lib.mod(v + 0.05, 1) * 10);
                 var msRounded = Math.round(v - msecTenths / 10);
-                ms = dateTime2ms(new Date(msRounded)) + msecTenths / 10;
+
+                var d = new Date(msRounded);
+                ms = dateTime2ms(d) + msecTenths / 10;
+                if(msUTC) {
+                    // For now it is only used
+                    // to fix bar length in milliseconds.
+                    // It could be applied in other places in v2
+                    ms += d.getTimezoneOffset() * ONEMIN;
+                }
```

Comment:
Can't we just (right after `v = +v`) say `if(msUTC) return v;` and skip the whole `dateTime2ms` step entirely in this case? Otherwise aside from the overhead there will be a subtle bug for very large negative values #2747

Reply:
Good call. Done in https://github.com/plotly/plotly.js/pull/4900/commits/2ae0fdfb53dfd3ce600347a2924f25793c4a7b5d.

Reply:
Actually the modified file was not saved. Let me push another commit...

Reply:
Fixed by https://github.com/plotly/plotly.js/pull/4900/commits/bb5bec34bd03214b4e026a6fc0f054379fd2f009.